File: library/src/main/kotlin/tag/type/version/GblVersionDependency.kt
package tag.type.version

import tag.GblType
import tag.Tag

data class GblVersionDependency(
    override val tagType: GblType,
    val imageType: ImageType,
    val statement: UByte,
    val reversed: UShort,
    val version: UInt,
): Tag {
    override fun copy(): Tag {
        return GblVersionDependency(
            tagType = tagType,
            imageType = imageType,
            statement = statement,
            reversed = reversed,
            version = version,
        )
    }
}

File: python-library/tag/type/version/gbl_version_dependency.py
"""
GBL Version Dependency tag implementation
Exact conversion from Kotlin GblVersionDependency.kt
"""

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ...gbl_type import GblType
    from ...tag import Tag

from .image_type import ImageType


# Імпорти з інших модулів (будуть додані пізніше):
# from ...tag import Tag


@dataclass
class GblVersionDependency:
    """GBL Version Dependency tag"""

    tag_type: 'GblType'
    image_type: ImageType
    statement: int  # UByte -> int
    reversed: int  # UShort -> int
    version: int  # UInt -> int

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblVersionDependency(
            tag_type=self.tag_type,
            image_type=self.image_type,
            statement=self.statement,
            reversed=self.reversed,
            version=self.version
        )